packages/analytics-controller/src/AnalyticsController.ts

```diff
@@ -0,0 +1,201 @@
+    optedIn = false,
+  }: AnalyticsControllerOptions) {
+    super({
+      name: controllerName,
+      metadata: controllerMetadata,
+      state: {
+        ...getDefaultAnalyticsControllerState(),
+        enabled,
+        optedIn,
+        ...state,
+      },
+      messenger,
+    });
+
+    this.#platformAdapter = platformAdapter;
+
+    // Register action handlers
+    this.#registerActionHandlers();
+  }
+
+  /**
+   * Register action handlers for messenger system
+   */
+  #registerActionHandlers(): void {
+    this.messenger.registerActionHandler(
+      `${controllerName}:trackEvent`,
+      this.trackEvent.bind(this),
+    );
+
+    if (this.#platformAdapter.identify) {
+      this.messenger.registerActionHandler(
+        `${controllerName}:identify`,
+        this.identify.bind(this),
+      );
+    }
+
+    if (this.#platformAdapter.trackPage) {
+      this.messenger.registerActionHandler(
+        `${controllerName}:trackPage`,
+        this.trackPage.bind(this),
+      );
+    }
+
+    this.messenger.registerActionHandler(
+      `${controllerName}:setEnabled`,
+      this.setEnabled.bind(this),
+    );
+
+    this.messenger.registerActionHandler(
+      `${controllerName}:setOptedIn`,
+      this.setOptedIn.bind(this),
+    );
+  }
+
+  /**
+   * Track an analytics event.
+   *
+   * Events are only tracked if analytics is enabled.
+   *
+   * @param eventName - The name of the event
+   * @param properties - Event properties
+   */
+  trackEvent(
+    eventName: string,
+    properties: AnalyticsEventProperties = {},
+  ): void {
+    // Don't track if analytics is disabled
+    if (!this.state.enabled) {
+      return;
+    }
+
+    // Delegate to platform adapter
+    this.#platformAdapter.trackEvent(eventName, properties);
+  }
+
+  /**
+   * Identify a user for analytics.
+   *
+   * @param userId - The user identifier (e.g., metametrics ID)
+   * @param traits - User traits/properties
+   */
+  identify(userId: string, traits?: AnalyticsEventProperties): void {
+    if (!this.state.enabled) {
+      return;
+    }
+
+    // Update state with analytics ID
+    this.update((state) => {
+      state.analyticsId = userId;
+    });
+
+    // Delegate to platform adapter if supported
+    if (this.#platformAdapter.identify) {
+      this.#platformAdapter.identify(userId, traits);
+    }
+  }
+
+  /**
+   * Track a page view.
+   *
+   * @param pageName - The name of the page
+   * @param properties - Page properties
+   */
+  trackPage(
+    pageName: string,
+    properties?: AnalyticsEventProperties,
+  ): void {
+    if (!this.state.enabled) {
+      return;
+    }
+
+    // Delegate to platform adapter if supported
+    if (this.#platformAdapter.trackPage) {
+      this.#platformAdapter.trackPage(pageName, properties);
+    }
+  }
+
+  /**
+   * Set the enabled state.
+   *
+   * @param enabled - Whether analytics tracking is enabled (default: true)
+   */
+  setEnabled(enabled: boolean = true): void {
```

Comment:
Since these implementations are so small, what are your thoughts on splitting these methods in to `enable`, `disable`, `optIn`, and `optOut`? The idea here is that methods should not be thought of as mere setters or getters but actions.